mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -883,12 +883,18 @@ def _get_orderby_clauses(order_by_list, session):
             if SearchUtils.is_metric(key_type, "="):
                 clauses.append(
                     sql.case(
-                        [(subquery.c.is_nan.is_(True), 1), (order_value.is_(None), 1)], else_=0
+                        [
+                            (subquery.c.is_nan == sqlalchemy.true(), 1),
+                            (order_value == sqlalchemy.null(), 1),
```

Comment:
"IS" is syntactically different from "=" (or "=="). See https://stackoverflow.com/questions/9822154/standard-sql-boolean-operator-is-vs-equals-operator
I'd prefer this line and L891 remain unchanged (I don't think it will cause a query error in mssql in its current form but please double check on your side).